tiledb/sm/enums/datatype.h

```diff
@@ -347,6 +352,31 @@ inline bool datatype_is_time(Datatype type) {
       type == Datatype::TIME_AS);
 }
 
+inline uint8_t datatype_to_underlying(Datatype type) {
+  return static_cast<uint8_t>(type);
+}
+
+/* Note: Valid enumeration values are between 0 and 40. */
+inline Status datatype_is_valid(Datatype type) {
```

Comment:
Needs doxygen comment

A function that returns a boolean, or boolean-like as this does, is better named as a predicate `is_datatype_valid` rather than an assertion `datatype_is_valid`.